app/controllers/user/degateway/post.controller.js

```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+const userService = require('../../../services/user.service')
+const paths = require('../../../paths')
+const { SHOW_DEGATEWAY_SETTINGS } = require('../../../utils/constants')
+
+module.exports = async function postDegatewayPreference (req, res, next) {
+  const degatewayPreference = req.body['degateway-preference']
+
+  try {
+    if (!SHOW_DEGATEWAY_SETTINGS) {
+      return res.redirect(paths.user.profile.index)
+    }
+    const { features } = await userService.findByExternalId(req.user.externalId)
+    // todo: should probably fix adminusers to take empty string for features rather than always having a value for subsequent patch requests
+    let featureSet = new Set(['default'])
+    features.forEach(feature => {
```

Comment:
Is `features` not going to be returned as a string, and then needs to be split into an array?

Reply:
see the [User constructor](https://github.com/alphagov/pay-selfservice/blob/bb00443e838cd218e32236d0db8d12465bfd7dc6/app/models/User.class.js#L47)